.github/workflows/main.yml

```diff
@@ -165,12 +165,41 @@ jobs:
       #       'halt\n' \
       #     | docker exec -i $CONTAINER iris session iris -UUSER
       #     docker stop $CONTAINER
+      - name: Clean install v0.7.x and test migration to latest.
+        timeout-minutes: 15
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          curl http://localhost:52773/registry/packages/-/all | jq
+          curl http://localhost:52773/registry/packages/zpm/ | jq
+          ASSET_NAME='zpm-0.7.1-beta.4.xml'
+          ASSET_URL=`wget --header "Authorization: token ${GITHUB_TOKEN}" -qO- https://api.github.com/repos/intersystems/ipm/releases | jq -r ".[].assets[] | select(.name == \"${ASSET_NAME}\") | .browser_download_url"`
+          wget $ASSET_URL -O /tmp/zpm.xml
+          docker run --network zpm --rm -d ${{ steps.image.outputs.name }} ${{ steps.image.outputs.flags }}
+          CONTAINER=`docker ps -lq`
+          docker cp /tmp/zpm.xml $CONTAINER:/home/irisowner/zpm.xml
+          docker cp tests/migration/v0.7-to-v0.9/. $CONTAINER:/tmp/test-package/
+          sleep 5; docker exec $CONTAINER /usr/irissys/dev/Cloud/ICM/waitISC.sh
+          /bin/echo -e '' \
```

Comment:
Personally I find using `heredoc` more readable and less error-prone. It doesn't require escaping line breaks (or potentially single quotes)

Reply:
It would look like this
```bash
docker exec -i $CONTAINER iris session iris -UUSER << EOF 
  set sc = ##class(%SYSTEM.OBJ).Load("/home/irisowner/zpm.xml", "ck")
  zpm "list":1
  zpm "repo -r -name registry -url ""http://registry:52773/registry/"" -username admin -password SYS":1
  zpm "repo -list"
  zpm "search":1
  zpm "install sslclient":1
  zpm "install zpm -v":1
  zpm "load -dev /tmp/test-package/":1
  zpm "test ipm-migration-v0.7-to-v0.9 -only -verbose":1
  halt
EOF
```